internal/shellgen/generate.go

```diff
@@ -50,6 +51,15 @@ func GenerateForPrintEnv(ctx context.Context, devbox devboxer) error {
 		return errors.WithStack(err)
 	}
 
+	if plan.needsGlibcPatch() {
+		patch, err := newGlibcPatchFlake(devbox.Config().NixPkgsCommitHash(), plan.Packages)
```

Comment:
Lets ship this. 

To flesh this out further, we should think more about how to enable configuring the `devbox.Config().NixPkgsCommitHash()` parameter. The field it uses is one we've been hoping to deprecate. Its not great to use a value that was there for a completely different reason or is used in a myriad of different use-cases, since users will not know what _other_ side-effects may arise if they change the commit-hash's value.

One suggestion is:
```
// devbox.json
{
    "meta-packages": {
           "patchGlibc": <glibc-installable> or `glibc@<version>`
     }
}
```
This would let users update it via `devbox update` as well.

Reply:
This was something that Mike and I discussed as an interim solution. I agree that it's not great how we're using it as a catch-all nixpkgs hash for random things.